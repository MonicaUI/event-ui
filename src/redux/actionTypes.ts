export const LOAD_ACCOUNT_SNAPSHOT = 'event-guestside-site/LOAD_ACCOUNT_SNAPSHOT';
export const LOAD_EVENT_SNAPSHOT = 'event-guestside-site/LOAD_EVENT_SNAPSHOT';
export const LOAD_WEBSITE_CONTENT = 'event-guestside-site/LOAD_WEBSITE_CONTENT';
export const SET_REFERRER = 'event-guestside-site/SET_REFERRER';
export const LOAD_REGISTRATION_PROCESS_CONTENT = 'event-guestside-site/LOAD_REGISTRATION_PROCESS_SNAPSHOT';
export const LOAD_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS';
export const LOAD_REG_CART_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_REG_CART_VISIBLE_PRODUCTS';
export const REPLACE_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/REPLACE_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE';
export const LOAD_EVENT_WEB_APP_SETTINGS = 'event-guestside-site/LOAD_EVENT_WEB_APP_SETTINGS';
export const SET_WEBSITE_PASSWORD_VALIDITY = 'event-guestside-site/SET_WEBSITE_PASSWORD_VALIDITY';
export const UPDATE_QUESTION_VISIBILTY = 'event-guestside-site/UPDATE_QUESTION_VISIBILITY';
export const GET_ORDERS = 'event-guestside-site/GET_ORDERS';
export const CLEAR_ORDERS = 'event-guestside-site/CLEAR_ORDERS';
export const SWITCH_CCPA_DIALOG_MODE = 'event-guestside-site/SWITCH_CCPA_DIALOG_MODE';
export const SHOW_TRANSPARENT_WRAPPER = 'event-guestside-site/SHOW_TRANSPARENT_WRAPPER';
export const HIDE_TRANSPARENT_WRAPPER = 'event-guestside-site/HIDE_TRANSPARENT_WRAPPER';
export const SPINNER_SELECTION_PENDING = 'event-guestside-site/SPINNER_SELECTION_PENDING';
export const SPINNER_SELECTION_DONE = 'event-guestside-site/SPINNER_SELECTION_DONE';
export const RESTORE_ALL_REG_TYPES_FOR_EVENT = 'event-guestside-site/RESTORE_ALL_REG_TYPES_FOR_EVENT';
