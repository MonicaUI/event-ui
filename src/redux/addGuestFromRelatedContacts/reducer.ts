const TOGGLE_RELATED_CONTACTS_VIEW = 'event-guestside-site/addGuestFromRelatedContacts/LIST_RELATED_CONTACTS';
const LOAD_RELATED_CONTACTS_FOR_INVITEE =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_FOR_INVITEE';
const LOAD_RELATED_CONTACTS_SEARCH_RESULTS =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_SEARCH_RESULTS';

export function setRelatedContacts(contactId: $TSFixMe, relatedContacts: $TSFixMe): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_FOR_INVITEE, payload: { contactId, relatedContacts } };
}

export function toggleRelatedContacts(): $TSFixMe {
  return { type: TOGGLE_RELATED_CONTACTS_VIEW };
}

export function setRelatedContactsSearch(
  contactId: $TSFixMe,
  searchCriteria: $TSFixMe,
  searchResults: $TSFixMe
): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_SEARCH_RESULTS, payload: { contactId, searchCriteria, searchResults } };
}

export default function addGuestFromRelatedContactsReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_RELATED_CONTACTS_FOR_INVITEE: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContacts: action.payload.relatedContacts
        }
      };
    }
    case TOGGLE_RELATED_CONTACTS_VIEW: {
      return {
        ...state,
        showRelatedContactsView: !(state as $TSFixMe).showRelatedContactsView
      };
    }
    case LOAD_RELATED_CONTACTS_SEARCH_RESULTS: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContactsSearchCriteria: action.payload.searchCriteria,
          relatedContactsSearchResults: action.payload.searchResults
        }
      };
    }
    default: {
      return state;
    }
  }
}
