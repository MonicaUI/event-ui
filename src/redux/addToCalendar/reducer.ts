import { CALENDAR_EVENT_ID } from '../../utils/addToCalendarUtils';

export const LOAD_ADD_TO_CALENDAR_PROVIDERS = 'event-guestside-site/calendar/LOAD_CALENDAR_PROVIDERS';
export const LOAD_EVENT_CALENDAR = 'event-guestside-site/calendar/LOAD_EVENT_CALENDAR';
export const LOAD_SESSION_CALENDAR = 'event-guestside-site/calendar/LOAD_SESSION_CALENDAR';

export function setAddToCalendarProviders(calendarProviders: $TSFixMe): $TSFixMe {
  return {
    type: LOAD_ADD_TO_CALENDAR_PROVIDERS,
    payload: { calendarProviders }
  };
}

export function setCalendarUrl(calendarType: $TSFixMe, calendarUrl: $TSFixMe, entityId = CALENDAR_EVENT_ID): $TSFixMe {
  if (entityId === CALENDAR_EVENT_ID) {
    return {
      type: LOAD_EVENT_CALENDAR,
      payload: { calendarType, calendarUrl }
    };
  }

  return {
    type: LOAD_SESSION_CALENDAR,
    payload: { calendarType, calendarUrl, entityId }
  };
}

export default function addToCalendarReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_ADD_TO_CALENDAR_PROVIDERS: {
      return {
        ...state,
        ...action.payload.calendarProviders
      };
    }
    case LOAD_EVENT_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [CALENDAR_EVENT_ID]: action.payload.calendarUrl
        }
      };
    }
    case LOAD_SESSION_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [action.payload.entityId]: action.payload.calendarUrl
        }
      };
    }
    default: {
      return state;
    }
  }
}
