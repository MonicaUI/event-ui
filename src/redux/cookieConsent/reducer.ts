// Reducer to update state for cookie Consent
import { DENY } from '../../utils/CookieConstants';

export const SET_COOKIE_CONSENT = 'event-guestside-site/cookieConsent/SET_COOKIE_CONSENT';

export default function cookieConsentReducer(state = { status: DENY }, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_COOKIE_CONSENT:
      return {
        ...state,
        status: action.payload.cookieStatus
      };
    default:
      return state;
  }
}
