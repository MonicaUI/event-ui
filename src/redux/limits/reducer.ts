const SET_ACCOUNT_LIMITS = 'event-guestside-site/limits/SET_ACCOUNT_LIMITS';

export const setAccountLimits = (accountLimits: $TSFixMe): $TSFixMe => {
  return {
    type: SET_ACCOUNT_LIMITS,
    payload: {
      accountLimits
    }
  };
};

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_ACCOUNT_LIMITS: {
      return action.payload.accountLimits;
    }
    default: {
      return state;
    }
  }
}
