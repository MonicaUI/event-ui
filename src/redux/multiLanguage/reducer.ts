import { SET_LOCALE } from 'nucleus-widgets/redux/modules/text';
import { LOAD_LANGUAGE } from './actions';

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_LOCALE:
      return {
        ...state,
        locale: action.payload.locale
      };
    case LOAD_LANGUAGE:
      return {
        ...state,
        loadedLanguages: [...state.loadedLanguages, action.payload]
      };
    default:
      return state;
  }
}
