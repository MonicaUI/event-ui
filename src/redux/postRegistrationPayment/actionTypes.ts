/* ACTION TYPES */

export const SUBMIT_WEB_PAYMENT = 'event-guestside-site/postRegistrationPayment/SUBMIT_WEB_PAYMENT';
export const RESET_SUBMIT_WEB_PAYMENT = 'event-guestside-site/postRegistrationPayment/RESET_SUBMIT_WEB_PAYMENT';
export const STORE_WEBPAYMENT_DATA = 'event-guestside-site/postRegistrationPayment/STORE_WEBPAYMENT_DATA';
export const FINALIZE_POSTREG_CHECKOUT_PENDING =
  'event-guestside-site/postRegistrationPayment/FINALIZE_POSTREG_CHECKOUT_PENDING';
export const FINALIZE_POSTREG_CHECKOUT_COMPLETE =
  'event-guestside-site/postRegistrationPayment/FINALIZE_POSTREG_CHECKOUT_COMPLETE';
