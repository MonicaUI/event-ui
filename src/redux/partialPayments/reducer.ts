import { UPDATE_PAYMENT_AMOUNT, UPDATE_SELECTED_PAYMENT_OPTION } from './actions';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';
import { START_MODIFICATION_SUCCESS } from '../registrationForm/regCart/actionTypes';
import { GET_ORDERS } from '../actionTypes';

// Reducer to update state for Partial Payments
export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  const defaultPartialPayment = { paymentAmountOption: null, paymentAmount: null };
  switch (action.type) {
    case UPDATE_SELECTED_PAYMENT_OPTION: {
      const FULL_PAYMENT_VALUE = PAYMENT_AMOUNT_OPTION.FULL_PAYMENT.value;
      const paymentOption =
        action.payload === FULL_PAYMENT_VALUE
          ? PAYMENT_AMOUNT_OPTION.FULL_PAYMENT
          : PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT;
      return {
        ...state,
        paymentAmountOption: paymentOption
      };
    }
    case UPDATE_PAYMENT_AMOUNT:
      return {
        ...state,
        paymentAmount: action.payload
      };
    case START_MODIFICATION_SUCCESS:
    case GET_ORDERS:
      return defaultPartialPayment;
    default:
      return state;
  }
}
