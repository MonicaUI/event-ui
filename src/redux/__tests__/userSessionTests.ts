import reducer, { setVerifiedWebsitePasswordInUserSession, getRegistrationTypeIdFromUserSession } from '../userSession';
import { LOG_OUT_REGISTRANT_SUCCESS, CLEAR_URL_IDENTIFYING_INFORMATION } from '../registrantLogin/actionTypes';
import createStoreWithMiddleware from 'nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware';
import { combineReducers } from 'redux';

const initialUserSessionUnknownInvitee = {
  eventId: 'b1cc65b3-af12-441d-9e8c-7681db58f96e',
  inviteeId: '608a5435-91c0-49eb-96e1-34f4c03599e2',
  isPreview: false,
  isPlanner: false,
  isTestMode: false,
  showEventBuildWizardBanner: false,
  eventBuildWizardExitUrl: '',
  licenseTypeId: 1,
  isFreeTrial: false,
  freeTrialPurchaseCta: '',
  httpReferrer: '',
  regTypeId: '',
  persistRegType: false
};

const initialUserSessionKnownInvitee = {
  ...initialUserSessionUnknownInvitee,
  inviteeId: '608a5435-91c0-49eb-96e1-34f4c03599e2'
};

const logoutAction = {
  type: LOG_OUT_REGISTRANT_SUCCESS,
  payload: {
    regTypeId: 'REG_TYPE_IN_ACTION_PAYLOAD',
    persistRegType: true
  }
};

const logoutActionWithExperimentOff = {
  type: LOG_OUT_REGISTRANT_SUCCESS
};

test('Verifying initial state.', () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

test('Unknown invitee LOG_OUT_REGISTRANT_SUCCESS should maintain regType in user session.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ eventId: string; inviteeId: st... Remove this comment to see the full error message
  const newState = reducer(initialUserSessionUnknownInvitee, logoutAction);
  expect(newState.regTypeId).toEqual('REG_TYPE_IN_ACTION_PAYLOAD');
  expect(newState.persistRegType).toBeTruthy();
});

test('Known invitee LOG_OUT_REGISTRANT_SUCCESS with persistRegType experiment off.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ inviteeId: string; eventId: st... Remove this comment to see the full error message
  const newState = reducer(initialUserSessionKnownInvitee, logoutActionWithExperimentOff);
  expect(newState.regTypeId).toEqual('');
  expect(newState.persistRegType).toBeFalsy();
});

test('Known invitee LOG_OUT_REGISTRANT_SUCCESS logs out invitee user.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ inviteeId: string; eventId: st... Remove this comment to see the full error message
  expect(reducer(initialUserSessionKnownInvitee, logoutAction)).toMatchSnapshot();
});

const clearFieldAction = {
  type: CLEAR_URL_IDENTIFYING_INFORMATION
};

test('Unknown invitee CLEAR_URL_IDENTIFYING_INFORMATION should clear regType in user session.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ eventId: string; inviteeId: st... Remove this comment to see the full error message
  const newState = reducer(initialUserSessionUnknownInvitee, clearFieldAction);
  expect(newState.regTypeId).toEqual('');
  expect(newState.persistRegType).toBeFalsy();
});

test('Known invitee CLEAR_URL_IDENTIFYING_INFORMATION should clear regType in user session.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ inviteeId: string; eventId: st... Remove this comment to see the full error message
  const newState = reducer(initialUserSessionKnownInvitee, clearFieldAction);
  expect(newState.regTypeId).toEqual('');
  expect(newState.persistRegType).toBeFalsy();
});

test('getRegistrationTypeIdFromUserSession', () => {
  const state = {
    userSession: {
      regTypeId: 'b1cc65b3-af12-441d-9e8c-7681db58f96e'
    }
  };

  expect(getRegistrationTypeIdFromUserSession(state)).toBe(state.userSession.regTypeId);
  expect(getRegistrationTypeIdFromUserSession({})).toBe('00000000-0000-0000-0000-000000000000');
});

describe('setVerifiedWebsitePasswordInUserSession', () => {
  let store;
  beforeEach(() => {
    store = createStoreWithMiddleware(
      combineReducers({
        website: (x = {}) => x,
        appData: (x = {}) => x,
        text: (x = {}) => x,
        clients: (x = {}) => x,
        visibleProducts: (x = {}) => x,
        registrationForm: (x = {}) => x,
        eventSnapshotVersion: (x = {}) => x,
        event: (x = {}) => x,
        timezones: (x = {}) => x,
        userSession: reducer
      })
    );
  });

  it('Sets setVerifiedWebsitePasswordInUserSession to true', () => {
    store.dispatch(setVerifiedWebsitePasswordInUserSession(true));
    expect(store.getState().userSession.verifiedWebsitePassword).toBeTruthy();
  });
  it('Sets setVerifiedWebsitePasswordInUserSession to false', () => {
    store.dispatch(setVerifiedWebsitePasswordInUserSession(false));
    expect(store.getState().userSession.verifiedWebsitePassword).toBeFalsy();
  });
});
