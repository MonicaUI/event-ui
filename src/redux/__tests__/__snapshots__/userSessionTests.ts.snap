// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Known invitee LOG_OUT_REGISTRANT_SUCCESS logs out invitee user. 1`] = `
Object {
  "persistRegType": true,
  "regTypeId": "REG_TYPE_IN_ACTION_PAYLOAD",
  "scrollToRegType": undefined,
  "verifiedWebsitePassword": undefined,
}
`;

exports[`Verifying initial state. 1`] = `
Object {
  "persistRegType": false,
  "regTypeId": "",
  "scrollToRegTyoe": false,
}
`;
